### PR TITLE
{vis}[foss/2023a] micro-sam v1.0.1, torch_em v0.7.1, affogato v0.3.3, ...

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/t/torch_em/torch_em-0.7.1-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,117 @@
+easyblock = 'PythonBundle'
+
+name = 'torch_em'
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/t/torch_em/torch_em-0.7.1-foss-2023a.eb]: fixed in a100374

File: easybuild/easyconfigs/n/nifty/nifty-1.2.1-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ('HDF5', '1.14.0'),
+]
+
+# Add flags (nifty/.github/workflows/build_unix.sh)
+configopts = '-DWITH_QPBO=OFF -DWITH_HDF5=ON -DWITH_Z5=OFF -DWITH_ZLIB=ON -DWITH_CPLEX=OFF -DWITH_GUROBI=OFF '
+# Fix path to python executable - it finds v3.6.8 from /usr/bin/ without this fix
+configopts += '-DPYTHON_EXECUTABLE="$EBROOTPYTHON/bin/python" -DBUILD_NIFTY_PYTHON=ON '
+
+# export PYTHON_MODULE_INSTALL_DIR to fix make install
+preinstallopts = 'export PYTHON_MODULE_INSTALL_DIR="%(builddir)s/easybuild_obj/python/" '
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/n/nifty/nifty-1.2.1-foss-2023a.eb]: fixed in a100374

File: easybuild/easyconfigs/z/z5py/z5py-2.0.17-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ('zarr', '2.17.1'),
+    ('pybind11', '2.11.1'),
+]
+
+configopts = '-DWITH_ZLIB=ON -DWITH_BZIP2=ON '
+# Fix path to python executable - it finds v3.6.8 from /usr/bin/ without this fix
+configopts += '-DPYTHON_EXECUTABLE="$EBROOTPYTHON/bin/python" '
+
+# export PYTHON_MODULE_INSTALL_DIR to fix make install
+preinstallopts = 'export PYTHON_MODULE_INSTALL_DIR="%(builddir)s/easybuild_obj/python/" '
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/z/z5py/z5py-2.0.17-foss-2023a.eb]: fixed in d58a1fc

File: easybuild/easyconfigs/t/torch_em/torch_em-0.7.1-foss-2023a.eb [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    ('xarray', '2023.9.0'),
+    ('affogato', '0.3.3'),
+]
+
+use_pip = True
+sanity_pip_check = True
+
+# avoid using hatchling as build backend (since installing it for this old Python version is a nightmare)
+# copied from MONAI-1.0.1-foss-2022a.eb
+local_preinstallopts_termcolor = ' '.join([
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/t/torch_em/torch_em-0.7.1-foss-2023a.eb]: fixed in d58a1fc

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in total)
node3107.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/61ff9c6881baf2f48224d152e4823166 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @pavelToman!

**Pull request merged**